conda/core/solve.py

```diff
@@ -2153,6 +2155,155 @@ def working_state_reset(self):
         self.final_environment_specs = None
 
 
+class TrackedMap(MutableMapping):
+    def __init__(self, name, data=None):
+        self._data = data
+        self._name = name
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+    def _set(self, key, value, *, reason=None, _level=3):
+        msg = f'{self._name}[{key}] set to `{value}`'
```

Comment:
I'd like to have finer debugging here. Whether the key was new or not; if it wasn't, what the value was and the new one is.